### PR TITLE
transfer-circuits: Remove output circuit variants

### PR DESCRIPTION
**Pull request by vlopes11**
The execute circuit variants increases the storage requirements of Rusk.
The variants can be reduced, depending only on the inputs.

The only check that is performed with the outputs is the homomorphic
sum, and this is trivial to be mocked via public inputs with zeroed
pedersen commitments.

This PR introduces a circuit padding with these special picked values.

Resolves https://github.com/dusk-network/rusk/issues/693

### CODE REVIEW

File: rusk/src/lib/services/prover.rs
Comment (herr-seppia) [inline rusk/src/lib/services/prover.rs:75]: Is this still triggered when `outputs>2`? Cannot find the check

Comment (vlopes11) [reply rusk/src/lib/services/prover.rs:75]: You're right, the check is missing. Good catch!

Comment (vlopes11) [reply rusk/src/lib/services/prover.rs:75]: https://github.com/dusk-network/rusk/pull/694/commits/eb80fcdef99ad9478d90ab974fb5d31319335963

File: rusk/src/lib/services/prover.rs
Comment (herr-seppia) [inline rusk/src/lib/services/prover.rs:83]: If the output Len is now fixed to 2, shouldn't this be `5 + inputs.len() + 2 * 2` => `9 + inputs.len()`?

**Review by herr-seppia — COMMENTED** [review]
Just last doubt before approval

File: rusk/src/lib/services/prover.rs
Comment (vlopes11) [reply rusk/src/lib/services/prover.rs]: https://github.com/dusk-network/rusk/pull/694/commits/4e39f0b16afc9243cceedf047ee71fafd98f3c4b

File: circuits/transfer/src/execute/output.rs
Comment (vlopes11) [reply circuits/transfer/src/execute/output.rs:73]: You are right, we are forced to use these magic values because we lack `const` functions. Blocked by: https://github.com/dusk-network/dusk-pki/pull/64/files

**Review by ureeves — CHANGES_REQUESTED** [review]
Lets apply the new `const fn`s in https://github.com/dusk-network/dusk-pki/pull/64 before merging this.

**Review by ureeves — APPROVED** [review]
LGTM. Would be nice for some commentary on the padding constants

**Pull request merged**
